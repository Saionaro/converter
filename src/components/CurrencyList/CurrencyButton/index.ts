export { CurrencyButton, TEST_ID } from "./CurrencyButton";
