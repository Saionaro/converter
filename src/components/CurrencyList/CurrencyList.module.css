.root {
  min-height: 100px;
}

.variants {
  font-size: 12px;
  font-weight: 200;
}

.list {
  margin-top: 8px;
}

.item + .item {
  margin-top: 8px;
}
