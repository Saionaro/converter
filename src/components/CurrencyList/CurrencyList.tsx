import { useStoreon } from "storeon/react";
import { CURRENCIES, Currency } from "src/constants/currencies";
import { WalletsStore } from "src/store/wallets";

import { CurrencyButton } from "./CurrencyButton";
import st from "./CurrencyList.module.css";
import { useCallback } from "react";

interface Props {
  onActivate: (val: Currency) => void;
  disabled: Partial<Record<Currency, boolean>>;
  activeCurrency?: Currency;
  disableMethod?: "hide" | "disable";
  showAmount?: boolean;
}

export function CurrencyList({
  onActivate,
  disabled,
  showAmount,
  activeCurrency,
  disableMethod = "hide",
}: Props) {
  const { wallets } = useStoreon<WalletsStore>("wallets");
  const handleClick = useCallback(
    (cur: Currency) => onActivate(cur),
    [onActivate]
  );

  return (
    <div className={st.root}>
      <h3 className={st.variants}>Wallets</h3>
      <ul className={st.list}>
        {CURRENCIES.map((cur) => {
          if (disabled[cur] && disableMethod === "hide") return null;
          return (
            <li key={cur} className={st.item}>
              <CurrencyButton
                currency={cur}
                onClick={handleClick}
                active={activeCurrency === cur}
                disabled={disabled[cur] && disableMethod === "disable"}
                amount={showAmount ? wallets[cur].amount : ""}
              />
            </li>
          );
        })}
      </ul>
    </div>
  );
}
