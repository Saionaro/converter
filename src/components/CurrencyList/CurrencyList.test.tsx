import React from "react";
import { render, fireEvent } from "@testing-library/react";

import { CurrencyList } from "./CurrencyList";

describe("components/CurrencyList", () => {
  test("fit snapshot", () => {
    const { container } = render(
      <CurrencyList onActivate={jest.fn()} disabled={{}} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("disableMethod: hides extra items", () => {
    const { container } = render(
      <CurrencyList
        onActivate={jest.fn()}
        disabled={{ USD: true }}
        disableMethod="hide"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("invokes callback with currency id", () => {
    const mockCallback = jest.fn();
    const { getByText } = render(
      <CurrencyList onActivate={mockCallback} disabled={{}} />
    );

    const usdButton = getByText("USD");
    fireEvent.click(usdButton);

    expect(mockCallback).toBeCalledTimes(1);
    expect(mockCallback).toBeCalledWith("USD");
  });
});
