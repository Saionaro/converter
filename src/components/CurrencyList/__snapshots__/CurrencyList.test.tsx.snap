// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CurrencyList disableMethod: hides extra items 1`] = `
<div>
  <h3>
    Wallets
  </h3>
  <ul>
    <li>
      <button
        class=""
        data-testid="currency-button"
      >
        EUR
      </button>
    </li>
    <li>
      <button
        class=""
        data-testid="currency-button"
      >
        GBP
      </button>
    </li>
  </ul>
</div>
`;

exports[`components/CurrencyList fit snapshot 1`] = `
<div>
  <h3>
    Wallets
  </h3>
  <ul>
    <li>
      <button
        class=""
        data-testid="currency-button"
      >
        USD
      </button>
    </li>
    <li>
      <button
        class=""
        data-testid="currency-button"
      >
        EUR
      </button>
    </li>
    <li>
      <button
        class=""
        data-testid="currency-button"
      >
        GBP
      </button>
    </li>
  </ul>
</div>
`;
